### PR TITLE
feat: list endpoints

### PR DESCRIPTION
**Pull request by lucasmarshall**
## Test Plan

Tested locally against hubspot and salesforce account and contact lists

### CODE REVIEW

File: apps/api/routes/crm/v2/lead.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default function init(app: Router): void {
       const snakecasedKeysLead = toSnakecasedKeysCrmLead(lead);
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       const { raw_data, ...rest } = snakecasedKeysLead;
-      return res.status(200).send(req.query.include_raw_data === 'true' ? snakecasedKeysLead : rest);
+      return res.status(200).send(req.query?.include_raw_data?.toString() === 'true' ? snakecasedKeysLead : rest);
```
Comment (asdfryan) [inline apps/api/routes/crm/v2/lead.ts:34]: refactor into util function? There's a bunch of places where we do this.

File: packages/core/remotes/impl/hubspot/index.ts [hunk 2]
```diff
@@ -2017,6 +2023,102 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
...
+    // Map response to ListMetadata interface
+    return {
+      results: response.data.lists.map((record: any) => ({
+        name: record.name,
+        label: record.name,
+        id: record.listId.toString(),
+        objectType: objectType,
+        rawData: record,
+      })),
+      totalCount: 0, // no way to get the total count from hubspot
```
Comment (asdfryan) [inline packages/core/remotes/impl/hubspot/index.ts:2057]: I think it's better to set the `totalCount` to be undefined and make it optional in the openAPI spec?

File: packages/core/remotes/impl/salesforce/index.ts [hunk 3]
```diff
@@ -1366,6 +1371,102 @@ ${modifiedAfter ? `WHERE SystemModstamp > ${modifiedAfter.toISOString()} ORDER B
...
+        id: record.Id,
+        objectType: objectType,
+        rawData: record,
+      })),
+      totalCount,
+      next: encodeCursor({
+        id: response.length ? response[response.length - 1].Id : null,
+        reverse: false,
+      }),
+      previous: cursor ? encodeCursor(cursor) : null,
```
Comment (asdfryan) [inline packages/core/remotes/impl/salesforce/index.ts:1412]: I don't think this is right? We're setting the previous cursor to be the previous one?

File: packages/core/remotes/impl/salesforce/index.ts [hunk 4]
```diff
@@ -1366,6 +1371,102 @@ ${modifiedAfter ? `WHERE SystemModstamp > ${modifiedAfter.toISOString()} ORDER B
...
+        id: listMetadata.Id,
+        objectType,
+        rawData: listMetadata,
+      },
+      totalCount,
+      next: encodeCursor({
+        id: (response.records.length ? response.records[response.records.length - 1].Id : null) as string,
+        reverse: false,
+      }),
+      previous: cursor ? encodeCursor(cursor) : null,
```
Comment (asdfryan) [inline packages/core/remotes/impl/salesforce/index.ts:1466]: Same here

File: packages/core/remotes/impl/salesforce/index.ts (hunk 3 above)
Comment (lucasmarshall) [reply packages/core/remotes/impl/salesforce/index.ts:1412]: Yeah, I thought I removed this. Will change.

File: packages/core/remotes/impl/hubspot/index.ts (hunk 2 above)
Comment (lucasmarshall) [reply packages/core/remotes/impl/hubspot/index.ts:2057]: Yeah, that's probably more reasonable

File: apps/api/middleware/openapi.ts [hunk 5]
```diff
@@ -38,6 +38,7 @@ export const openapiMiddleware = (specDir: string, version = 'v2') => {
     validateSecurity: false,
     validateRequests: {
       removeAdditional: true,
+      coerceTypes: true,
```
Comment (asdfryan) [inline apps/api/middleware/openapi.ts:41]: nice. Does this work for `read_from_cache={1|true|false|0}`?

File: apps/api/routes/crm/v2/account.ts [hunk 6]
```diff
@@ -35,7 +35,7 @@ export default function init(app: Router): void {
       req: Request<ListAccountsPathParams, ListAccountsResponse, ListAccountsRequest, ListAccountsQueryParams>,
       res: Response<ListAccountsResponse>
     ) => {
-      if (req.query?.read_from_cache?.toString() !== 'true') {
+      if (req.query?.read_from_cache !== true) {
```
Comment (asdfryan) [inline apps/api/routes/crm/v2/account.ts:38]: Nit: `!req.query?.read_from_cache`

File: apps/api/routes/crm/v2/account.ts [hunk 7]
```diff
@@ -50,7 +50,7 @@ export default function init(app: Router): void {
         pagination,
         records: records.map((record) => ({
           ...record,
-          raw_data: req.query.include_raw_data === 'true' ? record.raw_data : undefined,
+          raw_data: req.query?.include_raw_data === true ? record.raw_data : undefined,
```
Comment (asdfryan) [inline apps/api/routes/crm/v2/account.ts:53]: same here

`req.query?.include_raw_data ? ...`

**Review by asdfryan — APPROVED** [review]

**Comment by asdfryan** [comment]
Looks like there's some build failures?

```
#53 63.36 api:build: routes/marketing-automation/v2/forms.ts(78,21): error TS2367: This comparison appears to be unintentional because the types 'boolean | undefined' and 'string' have no overlap.
```

File: apps/api/middleware/openapi.ts (hunk 5 above)
Comment (lucasmarshall) [reply apps/api/middleware/openapi.ts:41]: AFAIK, yes.

**Review by asdfryan — APPROVED** [review]

**Pull request merged**